### DISCUSSION TITLE
best practice to integrate CONFIG variables in sysbuild for application only

### DISCUSSION THREAD

**Discussion by mathiasstorck** [zephyrproject-rtos/zephyr#61225] [Q&A] [answered]
Hi,

I'm starting using sysbuild to generate the mcuboot bootloader and the application at once.
For the application I behave different depending if the bootloader should be build or not.
So I have 2 options:

1. no bootloader, application only:

- west build my-application
- -> my-application/prj.conf is used only
- CONFIG_BOOTLOADER_MCUBOOT is not set
- CONFIG_MCUMGR_* should not be configured

2. bootloader + application

- west build my-application --sysbuild
- -> sysbuild/mcuboot.conf is used for additional config variables for mcuboot
- -> SB_CONFIG_BOOTLOADER_MCUBOOT is set there, CONFIG_BOOTLOADER_MCUBOOT will be set
- CONFIG_MCUMGR_* should be configured for main app only
- so I want to specify these additional config variables for the main application also now, something like sysbuild/application.conf
- these should be added to prj.conf

Which is the best practice to do so?

Thanks
Mathias

**Comment by mathiasstorck** [comment]
SB_CONFIG_BOOTLOADER_MCUBOOT is set in sysbuild.conf of course
additional config for mcuboot is set in sysbuild/mcuboot.conf

**Comment by mathiasstorck** [comment, marked as answer]
Just found out that I can introduce a sysbuild/app.conf (as my $APP_DIR is app) that is taken into account
Unfortunalty this is not documented, documentation should get improved there

**Comment by Sylphe88** [comment]
If anyone stumbles upon this, it is "quickly" mentioned here:
https://docs.zephyrproject.org/latest/build/sysbuild/index.html#sysbuild-file-suffix-support
